mep/common/models.py

```diff
@@ -90,16 +90,32 @@ class Meta:
     def dates(self):
         '''Date or date range as a string for display'''
 
-        # if no dates are set, return an empty string
-        if not self.start_year and not self.end_year:
+        if not self.start_year and not self.end_year: # no dates are set
             return ''
 
-        # if start and end year are the same just return one year
-        if self.start_year == self.end_year:
-            return self.start_year
+        if not self.end_year: # only start year
+            if self.start_year < 0: # and it's BCE
+                return str(abs(self.start_year)) + ' BCE-' # '100 BCE-'
+            return str(self.start_year) + '-' # '1900-
+
+        if not self.start_year: # only end year
+            if self.end_year < 0: # it's BCE
+                return '-' + str(abs(self.end_year)) + ' BCE' # '-100 BCE'
+            return '-' + str(self.end_year) # it's CE, '-1900'
+
+        if self.start_year == self.end_year: # they're the same year
+            if self.start_year < 0: # it's BCE
+                return str(abs(self.start_year)) + ' BCE' # '100 BCE'
+            return str(self.start_year) # it's CE, '1900'
+
+        if self.start_year < 0: # start date is BCE
```

Comment:
For this one, refactoring into a method to just convert dates to 'BCE' 'CE' strings will help too, I think. That way, you can just do:

```'%s - %s' % (self._stringify_date(self.start_year), self._stringify_date(self.end_year))``` without needing a lot of if statements to manage it.